### PR TITLE
Enable Portworx Storage Driver

### PR DESCRIPTION
**Pull request by adityadani**
This change
- Enables portworx storage driver through the kubernetes volume plugins API.
- Reverts commit - dcb5eef
- Removes the duplicate proto generation error.

### CODE REVIEW

**Comment by adityadani** [comment]
@deads2k I have modified the vendor'ed directories used by Portworx plugin to remove the "duplicate proto registered" error.

File: vendor/github.com/libopenstorage/openstorage/api/api.pb.go [hunk 1]
```diff
@@ -7,7 +7,7 @@ package api
 import proto "github.com/golang/protobuf/proto"
 import fmt "fmt"
 import math "math"
-import google_protobuf "go.pedge.io/pb/go/google/protobuf"
+import google_protobuf "github.com/golang/protobuf/ptypes/timestamp"
```
Comment (deads2k) [inline vendor/github.com/libopenstorage/openstorage/api/api.pb.go]: @smarterclayton this is a carry and I don't know what it does.  Do you?

File: Godeps/Godeps.json [hunk 2]
```diff
@@ -2535,27 +2535,27 @@
 		},
 		{
 			"ImportPath": "github.com/libopenstorage/openstorage/api",
-			"Rev": "6e787003b91ddba85f108b8aede075b1af0d3606"
+			"Rev": "a53f5e5662367da02b95470980c5dbaadfe96c99"
```
Comment (deads2k) [inline Godeps/Godeps.json]: Why did this bump if you have made a carry commit?  Did something else change?

Comment (adityadani) [inline Godeps/Godeps.json]: I might have understood the meaning of "carry" differently.
I updated this vendor'ed file to that particular commit. Should the commit message then be different?

File: vendor/github.com/libopenstorage/openstorage/api/api.pb.go (hunk 1 above)
Comment (adityadani) [inline vendor/github.com/libopenstorage/openstorage/api/api.pb.go]: Previously there were two different protobuf versions being used. I updated this vendor'ed code used by portworx to use the same protobuf version which others are using in openshift.

Comment (deads2k) [inline vendor/github.com/libopenstorage/openstorage/api/api.pb.go]: > Previously there were two different protobuf versions being used. I updated this vendor'ed code used by portworx to use the same protobuf version which others are using in openshift. Now we won't get the duplicate proto error.

Oh, ok.  Then this isn't a carry.  `<carry>` is a commit that updates code in a way that the upstream project rejected (or would reject if you know the project very well).  What you have done is a `bump`.  You should combine the code change the godep.json change into a single commit with a name like `bump(github.com/libopenstorage/openstorage): a53f5e5662367da02b95470980c5dbaadfe96c99`

Comment (deads2k) [inline vendor/github.com/libopenstorage/openstorage/api/api.pb.go]: > commit with a name like bump(github.com/libopenstorage/openstorage): a53f5e5662367da02b95470980c5dbaadfe96c99

We use that format to be able to find when we picked up particular levels to find particular patches.  A couple scripts rely on it.

Comment (adityadani) [inline vendor/github.com/libopenstorage/openstorage/api/api.pb.go]: I have updated the commits.

**Comment by adityadani** [comment]
@deads2k @smarterclayton Can you enable tests on this PR ?

**Comment by deads2k** [comment]
> I have updated the commits.

Thanks

/lgtm

**Comment by deads2k** [comment]
job died.  I'm going to cancel the rest to free the queue and retest

**Comment by adityadani** [comment]
The extended _conformance_install_update failed with this error :
Removing containers and the following error:
rm: cannot remove ‘/tmp/etcd/openshift-backup-etcd_backup_tag20170824144459’: Permission denied
@deads2k  Is this an intermittent error ?

**Comment by deads2k** [comment]
> The extended _conformance_install_update failed with this error :
> Removing containers and the following error:
> rm: cannot remove ‘/tmp/etcd/openshift-backup-etcd_backup_tag20170824144459’: Permission denied
> @deads2k Is this an intermittent error ?

it is :( .  We're battling a few flakes at the moment.  Looks like the bot got you sorted.

**Comment by adityadani** [comment]
@deads2k  I had to rebase my changes to fix some conflicts. Can you add the lgtm back? Also the flaky test failed again in the previous run.

**Comment by deads2k** [comment]
bumping queue priority since its getting stuck in rebases.

**Pull request merged**